### PR TITLE
Service 06

### PR DESCRIPTION
**Pull request by michalkocon**

### CODE REVIEW

**Comment by hcorg** [comment]
something is not building, any idea why? can you reproduce it locally?

**Comment by michalkocon** [comment]
Builds without any problem locally

File: service-06/Data.asn1 [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+--! You should have received a copy of the GNU General Public License
+--! along with this program.  If not, see <http://www.gnu.org/licenses/>.
+--!
+
+Data DEFINITIONS AUTOMATIC TAGS ::= BEGIN
+EXPORTS ALL;
+IMPORTS
+    PUSC-UINT32, PUSC-OCTSTR-VARIABLE-LEN FROM BasicTypes;
+
+MemoryData ::= SEQUENCE --TODO
```
Comment (hcorg) [inline service-06/Data.asn1]: why TODO?

File: service-06/Memory.asn1 [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+--!
+
+Memory DEFINITIONS AUTOMATIC TAGS ::= BEGIN
+EXPORTS ALL;
+IMPORTS
+    PUSC-UINT32, PUSC-OCTSTR-VARIABLE-LEN FROM BasicTypes;
+
+Memory-ID ::= ENUMERATED --TODO
+{
+    memid1 (0)
```
Comment (hcorg) [inline service-06/Memory.asn1]: `memId1` looks better :)

File: service-06/PUS-6-1.asn1 [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+TC-6-1-LoadObjectMemoryDataMultipleMemid ::= TC-6-1-LoadObjectMemoryDataGeneric {BaseMultipleMemid}
+
+LoadObjectInstruction ::= LoadObjectInstructionGeneric {PUSC-UINT32, MemoryData}
+
+TC-6-1-LoadObjectMemoryDataGeneric {Base-Type} ::= SEQUENCE
+{
+--! memory-ID Memory-ID-Type, moved to acn
+    base-ID Base-Type,
+--! n UnsignedInteger-Type, moved to acn
+    loadObjectInstructions SEQUENCE (SIZE(0 .. maxTC-6-1-LoadObjectInstructionCount)) OF LoadObjectInstruction
```
Comment (hcorg) [inline service-06/PUS-6-1.asn1]: can it be `0`?

Comment (hcorg) [reply service-06/PUS-6-1.asn1]: and shouldn't LoadObjectInsctuction be a generic parameter?

Comment (michalkocon) [reply service-06/PUS-6-1.asn1]: Yeah, should be > 1. 

LoadObjectInsctuction is constant part of this message. Only it's fields should be parameterized, not it's structure

Comment (hcorg) [reply service-06/PUS-6-1.asn1]: but that makes it possible parameter of current structure :) for example imagine system usng two different pus6 services for two different modules :)

**Review by hcorg — APPROVED** [review]

**Pull request merged**